src/main/java/net/contargo/types/telephony/validation/ValidPhoneNumberValidator.java

```diff
@@ -17,34 +17,27 @@
  * @author  Robin Jayasinghe - jayasinghe@synyx.de
  * @author  Olle Törnström - toernstroem@synyx.de
  */
-public class ValidPhoneNumberValidator implements ConstraintValidator<ValidPhoneNumber, PhoneNumber> {
+public class ValidPhoneNumberValidator implements ConstraintValidator<ValidPhoneNumber, String> {
 
     private static final int PHONE_NUMBER_SIZE = 64;
 
     @Override
     public void initialize(ValidPhoneNumber a) {
-
-        // nothing to do
     }
 
 
     @Override
-    public boolean isValid(PhoneNumber value, ConstraintValidatorContext cvc) {
+    public boolean isValid(String value, ConstraintValidatorContext cvc) {
 
-        // We must assume non-null, non-empty is validated elsewhere
-        if (value == null) {
+        if (StringUtils.isEmpty(value)) {
             return true;
         }
 
-        List<PhoneNumberValidationResult> validationResults = checkPhoneNumber(value);
+        final PhoneNumber phoneNumber = new PhoneNumber(value);
+        List<PhoneNumberValidationResult> validationResults = checkPhoneNumber(phoneNumber);
```

Comment:
I think this looks really nice overall. Have you thought about making `checkPhoneNumber` inline? Would that perhaps provide more clarity to the iteration of the validation results. Would it be an option to, otherwise, hide the constraint validation reporting?